phoenix-core/src/main/java/org/apache/phoenix/schema/stats/DefaultStatisticsCollector.java

```diff
@@ -121,92 +110,69 @@
-                Result result = htable.get(get);
-                if (!result.isEmpty()) {
-                    Cell cell = result.listCells().get(0);
-                    guidepostWidth = PLong.INSTANCE.getCodec().decodeLong(cell.getValueArray(), cell.getValueOffset(), SortOrder.getDefault());
-                } else if (!isViewIndexTable) {
-                    /*
-                     * The table we are collecting stats for is potentially a base table, or local
-                     * index or a global index. For view indexes, we rely on the the guide post
-                     * width column in the parent data table's metadata which we already tried
-                     * retrieving above.
-                     */
-                    try (Connection conn =
-                            QueryUtil.getConnectionOnServer(env.getConfiguration())) {
-                        PTable table = PhoenixRuntime.getTable(conn, tableName);
-                        if (table.getType() == PTableType.INDEX
-                                && table.getIndexType() == IndexType.GLOBAL) {
-                            /*
-                             * For global indexes, we need to get the parentName first and then
-                             * fetch guide post width configured for the parent table.
-                             */
-                            PName parentName = table.getParentName();
-                            byte[] parentKey =
-                                    SchemaUtil.getTableKeyFromFullName(parentName.getString());
-                            get = new Get(parentKey);
-                            get.addColumn(PhoenixDatabaseMetaData.TABLE_FAMILY_BYTES,
-                                PhoenixDatabaseMetaData.GUIDE_POSTS_WIDTH_BYTES);
-                            result = htable.get(get);
-                            if (!result.isEmpty()) {
-                                Cell cell = result.listCells().get(0);
-                                guidepostWidth =
-                                        PLong.INSTANCE.getCodec().decodeLong(cell.getValueArray(),
-                                            cell.getValueOffset(), SortOrder.getDefault());
-                            }
-                        }
-                    }
-
-                }
-            } finally {
-                if (htable != null) {
-                    try {
-                        htable.close();
-                    } catch (IOException e) {
-                        LOG.warn("Failed to close " + htable.getName(), e);
-                    }
-                }
-            }
+            long guidepostWidth = getGuidePostDepthFromSystemCatalog();
             if (guidepostWidth >= 0) {
                 this.guidePostDepth = guidepostWidth;
+                LOG.info("Guide post depth determined from SYSTEM.CATALOG: " + guidePostDepth);
             } else {
                 // Last use global config value
```

Comment:
This comment "// Last use global config value" isn't necessary any more as you have already commented the order of priority on the function.